### PR TITLE
add department management's business logic

### PR DESCRIPTION
**Pull request by hahunavth**
Link task: https://edu-redmine.sun-asterisk.vn/issues/68828
- [x] Update UI
- [x] Logic controller, services
- [x] Dùng file test để mock data

### CODE REVIEW

**Comment by hahunavth** [comment]
ready

**Comment by manhtd-1083** [comment]
ready

**Review by HuyyLinh — APPROVED** [review]

File: G4_DailyReport/src/main/java/com/example/G4_DailyReport/service/DepartmentService.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        return departmentRepository.findAll();
+    }
+
+    public Page<Department> findAll(Pageable pageable) {
+        Page<Department> pagedResult = departmentRepository.findAll(pageable);
+        if (pagedResult.hasContent()) {
+            return pagedResult;
+        } else {
+            return Page.empty();
+        }
```
Comment (vanvtt-0952) [inline G4_DailyReport/src/main/java/com/example/G4_DailyReport/service/DepartmentService.java]:
```suggestion
        if (pagedResult.hasContent()) {
            return pagedResult;
        }
        return Page.empty();
```

File: G4_DailyReport/src/main/java/com/example/G4_DailyReport/service/DepartmentService.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        Page<Department> pagedResult = departmentRepository.findAll(
+                spec,
+                pageable
+        );
+        if (pagedResult.hasContent()) {
+            return pagedResult;
+        } else {
+            return Page.empty();
+        }
```
Comment (vanvtt-0952) [inline G4_DailyReport/src/main/java/com/example/G4_DailyReport/service/DepartmentService.java]:
```suggestion
        if (pagedResult.hasContent()) {
            return pagedResult;
        }
        return Page.empty();
```

File: G4_DailyReport/src/main/resources/application.properties.example [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<<<<<<< HEAD
```
Comment (vanvtt-0952) [inline G4_DailyReport/src/main/resources/application.properties.example]: ...
fix conflict chưa xong bé bì nè

File: G4_DailyReport/src/main/resources/static/js/admin/jquery.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    // When the "Delete" button inside the modal is clicked, send the DELETE request
+    $('#confirmDeleteButton').click(function () {
+        var employeeId = $(this).data('department-id');
+        $.ajax({
+            url: '/admin/departments/' + employeeId,
+            type: 'DELETE',
+            success: function () {
+                // Reload the page after successful deletion
+                location.reload();
```
Comment (vanvtt-0952) [inline G4_DailyReport/src/main/resources/static/js/admin/jquery.js:17]: thêm cái mess lun emm, tiết kiệm ghê

**Pull request merged**